src/test/java/spoon/test/reference/TypeReferenceTest.java

```diff
@@ -510,15 +510,17 @@ public void testClearBoundsForTypeParameterReference() throws Exception {
 
 		reference.setBounds(null);
 
-		assertNull(reference.getBoundingType());
+		assertEquals(factory.Type().OBJECT, reference.getBoundingType());
+		assertTrue(reference.isDefaultBoundingType(reference.getBoundingType()));
 
 		reference.addBound(factory.Type().createReference(String.class));
 
 		assertNotNull(reference.getBoundingType());
 
 		reference.setBounds(new ArrayList<>());
 
-		assertNull(reference.getBoundingType());
+		assertEquals(factory.Type().OBJECT, reference.getBoundingType());
+		assertTrue(reference.isDefaultBoundingType(reference.getBoundingType()));
 	}
```

Comment:
add a test corresponding to the assert4j problem?

Reply:
> add a test corresponding to the assert4j problem?

I cannot reproduce exactly the assert4j problem, but we already have in our test a case with an explicit "? super Object", I changed the assert accordingly: https://github.com/INRIA/spoon/pull/1672/files#diff-c20710bb8ee877d0eae2d22578a8f010R506